src/lwc/boutiqueAppointments/boutiqueAppointments.js

```diff
@@ -150,4 +169,52 @@ export default class BoutiqueAppointments extends NavigationMixin(LightningEleme
                 break;
         }
     }
+
+    filterCasesByStatus(data) {
```

Comment:
MAJOR
This is `rookie` solution :)
Well, it works, but it is not optimal.

Use only one array instead - `this.data`. I`d say that you need only `data` and `allData` arrays.
consider using .filter() method to achieve that.